File: core/evidence/recursive_truth_boundary_engine.py
from __future__ import annotations

from typing import Any, Dict


def model_recursive_truth_boundaries(depth: int, evidence_count: int, *, min_evidence: int = 2) -> Dict[str, Any]:
    erosion = round(min(1.0, depth * 0.08), 3)
    bounded = evidence_count >= min_evidence and erosion < 0.5
    return {
        "bounded": bounded,
        "depth": depth,
        "erosion": erosion,
        "closure_allowed": False,
        "recursive_lock_in_allowed": False,
    }

File: src/evidence/recursiveTruthBoundaryEngine.ts
/**
 * Converted from Python: core/evidence/recursive_truth_boundary_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function modelRecursiveTruthBoundaries(depth: any, evidence_count: any, min_evidence: any = 2): any {
  var erosion: any = py.round(py.min([py.F(1.0), py.mul(depth, py.F(0.08))]), 3);
  var bounded: any = py.and2(py.ge(evidence_count, min_evidence), () => ((erosion < py.F(0.5))));
  return {"bounded": bounded, "depth": depth, "erosion": erosion, "closure_allowed": false, "recursive_lock_in_allowed": false};
}
